functions/webhooks/taskrouterCallback.protected.ts

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable global-require */
+/* eslint-disable import/no-dynamic-require */
+import '@twilio-labs/serverless-runtime-types';
+import { Context, ServerlessCallback } from '@twilio-labs/serverless-runtime-types/types';
+import { responseWithCors, bindResolve, success, error500 } from '@tech-matters/serverless-helpers';
+
+// eslint-disable-next-line prettier/prettier
+import type { AddCustomerExternalId } from '../private/addCustomerExternalId.protected';
+// eslint-disable-next-line prettier/prettier
+import type { PostSurveyJanitor } from '../private/postSurveyJanitor.protected';
+
+export type Body = {
+  EventType: string;
+  TaskSid?: string;
+  TaskAttributes?: string;
+};
+
+type EnvVars = {
+  TWILIO_WORKSPACE_SID: string;
+  CHAT_SERVICE_SID: string;
+  FLEX_PROXY_SERVICE_SID: string;
+};
+
+const TASK_CREATED_EVENT = 'task.created';
+const TASK_CANCELED_EVENT = 'task.canceled';
+const TASK_COMPLETED_EVENT = 'task.completed';
+
+const wait = (ms: number): Promise<void> => {
+  return new Promise((resolve) => {
+    setTimeout(resolve, ms);
+  });
+};
+
+export const handler = async (
+  context: Context<EnvVars>,
+  event: Body,
+  callback: ServerlessCallback,
+) => {
+  const response = responseWithCors();
+  const resolve = bindResolve(callback)(response);
+
+  const baseUrl = `https://${context.DOMAIN_NAME}`;
+
+  try {
+    const { EventType } = event;
+
+    if (EventType === TASK_CREATED_EVENT) {
+      const handlerPath = Runtime.getFunctions().addCustomerExternalId.path;
+      const addCustomerExternalId = require(handlerPath).addCustomerExternalId as AddCustomerExternalId;
+      await addCustomerExternalId(context, event);
```

Comment:
I remember we're were having issues like `calling 'path' of undefined`.

If this is still a potential issue, I feel like this `taskrouterCallback` could be split into two separate callbacks to avoid this workaround to invoke other functions `Runtime.getFunctions().addCustomerExternalId.path`:

1. TASK_CREATED_EVENT:  addCustomerExternalId
2. TASK_CANCELED_EVENT || TASK_COMPLETED_EVENT: postSurveyJanitor

If that's not an issue anymore, nevermind.